tests/test_builders/test_build_gettext.py

```diff
@@ -120,9 +124,9 @@ def test_msgfmt(app):
     srcdir='gettext',
     confoverrides={'gettext_compact': False},
 )
-def test_gettext_index_entries(app):
+def test_gettext_index_entries(app: SphinxTestApp) -> None:
     # regression test for #976
-    app.build(filenames=[app.srcdir / 'index_entries.txt'])
+    app.build(filenames=[str(app.srcdir / 'index_entries.txt')])
```

Comment:
`app.build` takes strings for filenames.

Reply:
i agree this is the right approach now, but `sphinx.builders.Builder.build_specific` should be refactored to accept an `Iterable[Path]` rather than `list[str]` in a separate PR